spec/message.abnf

```diff
@@ -0,0 +1,52 @@
+message = [s] *(declaration [s]) body [s]
+
+declaration = "let" s variable [s] "=" [s] "{" [s] expression [s] "}"
```

Comment:
That's a good point. My plan was to file a follow-up for us to decide whether we want `let` *and* `LET`, or perhaps just the lowercase `let`. We can then use RFC 7405's case-sensitive literals, e.g. `%s"let"` and `%s"LET"` to encode the decision in the grammar.

Reply:
[RFC 7405](https://datatracker.ietf.org/doc/html/rfc7405) defines a case sensitive variant of ABNF. Instead of `"let"` as a string literal you would use `%s"let"` to define it as case sensitive or `%i"let"` for insensitive. Conversion to RFC 5234 readable syntax should be pretty easy in reverse if you were to write RFC 7405 in the first place.

Reply:
Oops jinxed! My comment was started before the other reply, then I got sidetracked putting kids to bed.

Reply:
I filed #349 for this.

Reply:
If one or two words turn out to me the only case sensitive bits, you might be able to skip the extensions with just an exact character code match for the lower case letters:

```abnf
let = %d154.145.164
```

Reply:
Ah, good idea. Yes, there are only three keywords: `let`, `match`, and `when`.

Reply:
> That's a good point. My plan was to file a follow-up for us to decide whether we want `let` _and_ `LET`, or perhaps just the lowercase `let`. We can then use RFC 7405's case-sensitive literals, e.g. `%s"let"` and `%s"LET"` to encode the decision in the grammar.

New plan: the W3C flavor of EBNF that this PR migrates _from_ is case-sensitive, which means that the currently agreed upon grammar only accepts lowercase `let`, `match`, and `when`. I filed #349 to discuss possibly relaxing this, but in the meantime I added one more commit to this PR (https://github.com/unicode-org/message-format-wg/pull/347/commits/84bbfa1dd751eb1915514cf7eb37e3834748bbf8) which explicitly encodes the lowercase keywords in the ABNF, to maintain the status quo.